restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -911,18 +911,40 @@ void addRowsToMap(Map<String, List<Row>> rowsByDoc, List<Row> rows) {
     LinkedHashSet<String> existsByDoc = new LinkedHashSet<>();
     LinkedHashMap<String, Integer> countsByDoc = new LinkedHashMap<>();
 
+    List<FilterCondition> inCassandraFilters =
+        filters.stream()
+            .filter(f -> !FilterOp.LIMITED_SUPPORT_FILTERS.contains(f.getFilterOp()))
+            .collect(Collectors.toList());
+    List<FilterCondition> inMemoryFilters =
+        filters.stream()
+            .filter(f -> FilterOp.LIMITED_SUPPORT_FILTERS.contains(f.getFilterOp()))
+            .collect(Collectors.toList());
+
+    Set<String> inCassandraPaths =
+        inCassandraFilters.stream().map(f -> f.getFullFieldPath()).collect(Collectors.toSet());
+    if (inCassandraPaths.size() > 1 || inMemoryFilters.size() > 0) {
+      // If the request involves more than one filter that could be handled by cassandra, or if
+      // there is a combination of supported and unsupported filters, then we have to default to the
+      // I/O intensive
+      // in-memory filter.
+      inCassandraFilters = new ArrayList<>();
```

Comment:
nit: `Collections.emptyList()`?

Reply:
Yea thanks, I always forget about that :P